pkg/retriever/retriever.go

```diff
@@ -90,26 +90,29 @@ func NewRetriever(
 	config RetrieverConfig,
 	client RetrievalClient,
 	candidateFinder CandidateFinder,
+	bitswapRetriever types.CandidateRetriever,
 ) (*Retriever, error) {
 	retriever := &Retriever{
 		config:       config,
 		eventManager: events.NewEventManager(ctx),
 		spTracker:    newSpTracker(nil),
 	}
+	candidateRetrievers := []types.CandidateRetriever{
+		&GraphSyncRetriever{
+			GetStorageProviderTimeout: retriever.getStorageProviderTimeout,
+			IsAcceptableQueryResponse: retriever.isAcceptableQueryResponse,
+			Client:                    client,
+		},
+	}
+	if bitswapRetriever != nil {
+		candidateRetrievers = append(candidateRetrievers, candidateRetrievers...)
+	}
 	retriever.executor = combinators.RetrieverWithCandidateFinder{
 		CandidateFinder: NewAssignableCandidateFinder(candidateFinder, retriever.isAcceptableStorageProvider),
 		CandidateRetriever: combinators.SplitRetriever{
-			CandidateSplitter: NewProtocolSplitter([]multicodec.Code{multicodec.TransportGraphsyncFilecoinv1, multicodec.TransportBitswap}),
-			CandidateRetrievers: []types.CandidateRetriever{
-				&GraphSyncRetriever{
-					GetStorageProviderTimeout:   retriever.getStorageProviderTimeout,
-					IsAcceptableStorageProvider: retriever.isAcceptableStorageProvider,
-					IsAcceptableQueryResponse:   retriever.isAcceptableQueryResponse,
-					Client:                      client,
-				},
-				NewBitswapRetriever(),
-			},
-			CoordinationKind: types.RaceCoordination,
+			CandidateSplitter:   NewProtocolSplitter([]multicodec.Code{multicodec.TransportGraphsyncFilecoinv1, multicodec.TransportBitswap}),
+			CandidateRetrievers: candidateRetrievers,
```

Comment:
still not a fan of how these two arrays have to match up for this to work

Reply:
and, what's going to happen when our `candidateRetrievers` only has graphsync but the splitter is considering both? Do we need a `NullRetriever()` to take its place or something?

Reply:
no it's setup properly to just ignore